src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -708,19 +708,33 @@ class ColumnReferenceSegment(ObjectReferenceSegment):  # type: ignore
     )
 
     def extract_possible_references(self, level):
-        """Extract possible references of a given level."""
+        """Extract possible references of a given level.
+
+        Overrides the parent-class function. BigQuery's support for things like
+        the following
+        - Functions that take a table as a parameter (e.g. TO_JSON_STRING)
+          https://cloud.google.com/bigquery/docs/reference/standard-sql/
+          json_functions#to_json_string
+        - STRUCT
+
+        mean that, without schema information (which SQLFluff does not have),
+        references to data are often ambiguous.
+        """
         level = self._level_to_int(level)
         refs = list(self.iter_raw_references())
         if level == self.ObjectReferenceLevel.SCHEMA.value and len(refs) >= 3:
             return [refs[0]]  # pragma: no cover
-        if level == self.ObjectReferenceLevel.TABLE.value and len(refs) >= 3:
-            # Ambiguous case: The table could be the first or second part, so
-            # return both.
-            return [refs[0], refs[1]]
-        if level == self.ObjectReferenceLevel.OBJECT.value and len(refs) >= 3:
+        if level == self.ObjectReferenceLevel.TABLE.value:
+            # One part: Could be a table, e.g. TO_JSON_STRING(t)
+            # Two parts: Could be schema+table or table+column.
```

Comment:
I would change schema to dataset. Schema means something else in BigQuery:

```suggestion
            # Two parts: Could be dataset+table or table+column.
```

Reply:
Will do